### PR TITLE
task: Make Inactive users an enterprise feature

### PR DESCRIPTION
**Pull request by chriswk**
Ivar pointed out to me that this was intended as an enterprise only feature. So this PR makes it an enterprise only feature. Conditionally render the link in the normal user table, and use premium feature component if you happen to hit the route and not be running on the enterprise plan.

### CODE REVIEW

**Comment by chriswk** [comment]
> Linting changes makes it a bit harder to review this PR. Can we fix those first?

yes, it is on my TODO list :)

File: frontend/src/component/admin/users/InactiveUsersList/InactiveUsersList.tsx [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
+import {DeleteInactiveUsers} from './DeleteInactiveUsers/DeleteInactiveUsers';
+import {Link} from 'react-router-dom';
+import {StyledUsersLinkDiv} from '../Users.styles';
+import useUiConfig from "../../../../hooks/api/getters/useUiConfig/useUiConfig";
 
 export const InactiveUsersList = () => {
     const { removeUser, userApiErrors } = useAdminUsersApi();
     const { deleteInactiveUsers, errors: inactiveUsersApiErrors } =
         useInactiveUsersApi();
+    const { uiConfig, isPro, isEnterprise } = useUiConfig();
```
[the remark this answers is not in the document]
Comment (chriswk) [reply frontend/src/component/admin/users/InactiveUsersList/InactiveUsersList.tsx:36]: I started here, and realised I wanted to do it outside of the component, so these are unused. Thanks for reminding me. I'll clean up.

File: frontend/src/component/common/PremiumFeature/PremiumFeature.tsx [hunk 2]
```diff
@@ -123,6 +123,11 @@ const PremiumFeatures = {
         url: '', // FIXME: url
         label: 'Dashboard',
     },
+    'inactive-users': {
+        plan: FeaturePlan.ENTERPRISE,
+        url: '',
```
[the remark this answers is not in the document]
Comment (chriswk) [reply frontend/src/component/common/PremiumFeature/PremiumFeature.tsx:128]: Good question, we probably should. Not as part of this PR, but yes, we should absolutely write some docs for this.

File: frontend/src/component/admin/users/InactiveUsersList/InactiveUsersList.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chriswk) [reply frontend/src/component/admin/users/InactiveUsersList/InactiveUsersList.tsx]: Done in 34a7c6d

**Review by nunogois — APPROVED** [review]
LG! Two notes:
 - The backend code should be moved to Enterprise if possible, otherwise this is not really an Enterprise feature
 - I noticed a `showInactiveUsers` feature flag in our code base that has nothing to do with this feature. Not related with this PR, but it confused me in case anyone else stumbles upon it 😄

**Comment by chriswk** [comment]
> LG! Two notes:
> 
>     * The backend code should be moved to Enterprise if possible, otherwise this is not really an Enterprise feature
> 
>     * I noticed a `showInactiveUsers` feature flag in our code base that has nothing to do with this feature. Not related with this PR, but it confused me in case anyone else stumbles upon it 😄

Good input both points. I'm not sure what we want to do for OSS here tbh, it's not like you can't write something similar yourself. I think for now, I'm fine with this being guarded frontend only. Moving the backend can be set as a new task.

**Pull request merged**